tiny-dnn/tiny-dnn
# Xtensor integration (#739)

* Add xtensor 0.9.0

* Change C++ version to C++14

* Change parameters to xarray

* Replace all parameters with xtensors in FC

* Return values

* Replace with more general templates

* Testing

* Update xtensor to 0.10.1

* Fix segfault

* gcc version for tests

* clang version for tests

* fix wrong indexing

* fix memory access problem

* check if has bias

* remove broken source

* raw data usage

* use raw data

* Add templated checks for value_type and xexpressions.

* Fix biases pass

* clang-format

* clang-format

* typos

* clang-format

* return pass-by-value (temp)

* clang-format

* fix AVX segfault (#2)

* Start reviewing Tensor

* Cotinue Tensor refactoring

* Rewrite subview (still doesn't compile)

* Code fixes. Temporary workaround for views

* Remove tensor_storage

* Fix errors

* Add missing include, fix bugs

* Fixes of tests

* Fix docs, implementations, rewrite tests

* Fix bugs

* Fix formatting

* Variadic template version of view

* Remove dead code. Add test case

* Revert fully connected changes

* Typos

* Fix lack of support on tensors with different storage types. Add exponent test

* 1x1x1 tensor probably makes no sense

* OS X xcode8 for Travis

* OS X xcode8 for Travis

* OS X xcode8 for Travis

* Fix typos and code style

* Add dim() function, small fixes

* C++11 -> C++14

* Remove specific class from constructors. Remove resize function

* Fix wrong naming

* Use member initialization for constructors
